sqlobject/src/main/java/org/jdbi/v3/sqlobject/TransactionDecorator.java

```diff
@@ -21,23 +21,25 @@
 import org.jdbi.v3.core.transaction.TransactionCallback;
 import org.jdbi.v3.core.transaction.TransactionIsolationLevel;
 
-class PassThroughTransactionHandler implements Handler
+class TransactionDecorator implements Handler
 {
+    private final Handler delegate;
     private final TransactionIsolationLevel isolation;
-    private final PassThroughHandler delegate;
 
-    PassThroughTransactionHandler(Transaction tx)
+    TransactionDecorator(Handler delegate, Transaction tx)
     {
+        this.delegate = delegate;
         this.isolation = tx.value();
-        this.delegate = new PassThroughHandler();
     }
 
     @Override
     public Object invoke(Supplier<Handle> handle, SqlObjectConfig config, Object target, Object[] args, Method method) throws Exception
     {
         Handle h = handle.get();
+
         if (h.isInTransaction()) {
-            throw new TransactionException("Nested @Transaction detected - this is currently not supported.");
+            // Already in transaction. The outermost @Transaction method determines the transaction isolation level.
```

Comment:
I'm not sure I can do that.

The default transaction isolation level can vary by database vendor, so I can't hardcode a specific level to expect as the "default" in a nested call to bare `@Transaction` methods.